package.json

```diff
@@ -27,8 +27,12 @@
     "lint:js": "eslint ./*.js addon addon-test-support app blueprints config lib server test-support tests",
     "start": "ember serve",
     "test": "ember test",
-    "nodetest": "mocha node-tests --recursive",
-    "test:all": "ember try:each"
+    "ci:prepare": "yarn prepublishOnly && rimraf ts",
+    "ci:test": "ember test && mocha --recursive js/tests",
+    "test:node": "mocha -r register-ts-node 'ts/tests/**/*.{ts,js}'",
+    "test:all": "ember try:each",
+    "prepublishOnly": "yarn tsc --project ts --noEmit false",
+    "postpublish": "rimraf js"
```

Comment:
:+1: This will help tremendously with:
> This could lead to a pretty frustrating DX if one of us happens to have files sitting around in js/ and doesn't realize it.